drivers/timer/CMakeLists.txt

```diff
@@ -32,3 +32,10 @@ zephyr_library_sources_ifdef(CONFIG_SAM0_RTC_TIMER sam0_rtc_timer.c)
 zephyr_library_sources_ifdef(CONFIG_STM32_LPTIM_TIMER stm32_lptim_timer.c)
 zephyr_library_sources_ifdef(CONFIG_XLNX_PSTTC_TIMER xlnx_psttc_timer.c)
 zephyr_library_sources_ifdef(CONFIG_XTENSA_TIMER xtensa_sys_timer.c)
+
+if (CONFIG_NRF_GRTC_TIMER)
+	zephyr_library_sources(nrf_grtc_timer.c)
+	if (CONFIG_NRF_GRTC_TIMER_CLOCK_MANAGEMENT)
+		zephyr_library_compile_definitions(NRF_GRTC_HAS_EXTENDED=1)
+	endif()
+endif()
```

Comment:
this should likely be moved to `modules/hal_nordic` as an injected definition for HAL (as we do for other cases)